src/Controller.php

```diff
@@ -609,11 +612,21 @@ public static function wp2staticProcessQueue() : void {
                         WsLog::l( 'No deployment add-ons are enabled, skipping deployment.' );
                     } else {
                         WsLog::l( 'Starting deployment' );
-                        do_action(
-                            'wp2static_deploy',
-                            ProcessedSite::getPath(),
-                            $deployer
-                        );
+                        $query = "SELECT GET_LOCK('wp2static_jobs_deploying', 30) AS lck";
+                        $locked = intval( $wpdb->get_row( $query )->lck );
+                        if ( ! $locked ) {
+                            WsLog::l( 'Failed to acquire "wp2static_jobs_deploying" lock.' );
+                            return;
```

Comment:
Will this still let registered post deployment hooks fire? ie, if S3 deploy fails and we return here, might it still call CF invalidation unnecessarily?

Reply:
No, we're not catching the exceptions so it will not run post-deploy hooks in that case.